### DISCUSSION TITLE
Why Does useQuery Fetch Again on the Client Even with the Same Query Key?

### DISCUSSION THREAD

**Discussion by quantumgravity-co-kr** [TanStack/query#8379] [Q&A]
<img width="486" alt="스크린샷 2024-12-02 오후 4 30 44" src="https://github.com/user-attachments/assets/a72dec92-0832-4f8f-b7c6-94792d874824">

![스크린샷 2024-12-02 오후 4 24 27](https://github.com/user-attachments/assets/10c422e2-8ae5-4039-9500-917de99e8d38)

<img width="1171" alt="스크린샷 2024-12-02 오후 4 24 51" src="https://github.com/user-attachments/assets/e1bb57fd-a3ab-4846-b844-65258b43ddb3">

	•	Using the Same Query Key
	•	Confirmed that the prefetched value is correctly passed.
	•	However, why does useQuery attempt to fetch again on the client?

The staleTime is set to 0.
Is this the expected behavior? Shouldn’t the client avoid fetching if the query was already executed on the server with the same key?

To ensure fetching only happens on the server, do I need to use getQueryData instead?

**Comment by TkDodo** [comment]
because `staleTime` defaults to zero, so you get an automatic refetch on the client. Customizing `staleTime` is very likely what you want:

- https://tanstack.com/query/v5/docs/framework/react/guides/important-defaults
- https://tanstack.com/query/v5/docs/framework/react/guides/caching
- https://tkdodo.eu/blog/react-query-as-a-state-manager
- https://tkdodo.eu/blog/thinking-in-react-query
